veps/vep-4.md

```diff
@@ -0,0 +1,342 @@
+				log.Warningf("Use of single-dash long flags is deprecated and will be removed in 14.0. Please use --%s instead", f.Name)
+			}
+		}
+	})
+}
+
+// then replace all calls to flag.Parse() with a call to parse()
+```
+
+Which, when running the same `vtctl` binary, produces output like:
+
+```
+W1031 13:17:56.497111   88580 servenv.go:270] Use of single-dash long flags is deprecated and will be removed in 14.0. Please use --backup_storage_implementation instead
+W1031 13:17:56.497648   88580 servenv.go:270] Use of single-dash long flags is deprecated and will be removed in 14.0. Please use --cell instead
+```
+
+The second issue with dropping in `pflag` is that it supports mixing of positional and flag arguments.
+**This change will break subcommands, such as those in `vtctl` and `mysqlctl`.**
+
+To illustrate the problem, consider the command `./mycmd -a 1 -b 2 X -c 3 -d 4`.
+With `flag`, parsing this will result in two flags, `a` and `b`, `flag.NArg()` will return 5 and `flag.Args()` will return the slice `["X", "-c", "3", "-d", "4"]` ([c.f.](https://play.golang.org/p/sO4UlQPGg4l)).
+With `pflag`, all 4 flags will be parsed, `pflag.NArg()` will return 1, and `pflag.Args()` will return the slice `["X"]`.
+
+In order to match this behavior, users should update their commands to insert a double-dash (`--`) before the first positional argument.
+This causes `pflag` to stop looking for flags and treat everything after as a positional argument (matching what `flag` does even without the double-dash), and has no effect on the parsing behavior of `flag` ([c.f.](https://play.golang.org/p/76FrFrLFOMT)).
+
+As with the single-dash long flag issue, we can make code changes similar to the `parse` snippet above to warn users and target v14 for switching, or we can announce and move to `pflag` immediately.
+
+### Phase 2: `pflag` adoption and migration
+
+After the preparatory work we did in phase 1, dropping in `pflag` is fairly straightforward.
+For example, `servenv` has two functions, `ParseFlags` and `ParseFlagsWithArgs`.
+Taking the first as an example, we would write:
+
+```go
+import (
+    goflag "flag"
+    flag "github.com/spf13/pflag" // minimize code changes
+)
+func ParseFlags(cmd string) {
+    flag.CommandLine.AddGoFlagSet(goflag.CommandLine) // (TODO) Remove after all other packages add flags directly to pflag.
+	flag.Parse()
+
+	if *Version {
+		AppVersion.Print()
+		os.Exit(0)
+	}
+
+	args := flag.Args()
+	if len(args) > 0 {
+		flag.Usage()
+		log.Exitf("%s doesn't take any positional arguments, got '%s'", cmd, strings.Join(args, " "))
+	}
+}
+```
+
+Now, other packages which declare flags can begin moving to `pflag`.
+Once all packages are adding their flags directly to the `pflag.CommandLine` flagset, we can remove the "TODO" interop line.
+Now, other packages can gradually move over to `pflag`, and once everything is adding their flags directly to the `pflag` flagset, we can remove the interop line.
```

Comment:
Is there some duplicated text here that should be cleaned up?

Reply:
yeeeeeeep 😅